lib/auth/auth_with_roles.go

```diff
@@ -5391,6 +5391,60 @@ func (a *ServerWithRoles) ListReleases(ctx context.Context) ([]*types.Release, e
 	return a.authServer.releaseService.ListReleases(ctx)
 }
 
+// ListSAMLIdPServiceProviders returns a paginated list of SAML IdP service provider resources.
+func (a *ServerWithRoles) ListSAMLIdPServiceProviders(ctx context.Context, pageSize int, nextToken string) ([]types.SAMLIdPServiceProvider, string, error) {
```

Comment:
Is this a paginated API instead of a streaming API to accomodate the web UI? If not, I'd rather have a new streaming API (like `GetInstances`) rather than yet another paginated API that ends up being called in a loop every time anyway.